packages/jupyterlab-chat-extension/src/chat-commands/providers/user-mention.tsx

```diff
@@ -27,55 +27,78 @@ export const mentionCommandsPlugin: JupyterFrontEndPlugin<void> = {
 class MentionCommandProvider implements IChatCommandProvider {
   public id: string = 'jupyter-chat:mention-commands';
 
-  // regex used to test the current word
-  private _regex: RegExp = /^@[\w-]*:?/;
+  // Regex used to find all mentions in a message.
+  // IMPORTANT: the global flag must be specified to find >1 mentions.
+  private _regex: RegExp = /@[\w-]*:?/g;
 
-  async getChatCommands(inputModel: IInputModel) {
-    this._users.clear();
+  /**
+   * Lists all valid user mentions that complete the current word.
+   */
+  async listCommandCompletions(inputModel: IInputModel) {
+    // Return early if the current word does not match the expected syntax
     const match = inputModel.currentWord?.match(this._regex)?.[0];
     if (!match) {
       return [];
     }
 
-    const users = inputModel.chatContext.users;
-    users.forEach(user => {
-      let mentionName = user.mention_name;
-      if (!mentionName) {
-        mentionName = Private.getMentionName(user);
-        user.mention_name = mentionName;
-      }
-
-      this._users.set(mentionName, {
-        user,
-        icon: <Avatar user={user} />
-      });
-    });
-
-    // Build the commands for each user.
-    const commands: ChatCommand[] = Array.from(this._users)
-      .sort()
+    // Otherwise, build the list of `@`-mention commands that complete the
+    // current word.
+    const existingMentions = new Set(inputModel.value?.match(this._regex));
+    const commands: ChatCommand[] = Array.from(this._getUsers(inputModel))
+      // remove users whose mention names that do not complete the match
       .filter(user => user[0].toLowerCase().startsWith(match.toLowerCase()))
+      // remove users already mentioned in the message
+      .filter(user => !existingMentions.has(user[0]))
```

Comment:
Maybe another way could be to change the order, instead of removing the suggestion.
The users already mentioned would be on top of the list, but would still be available for a second mention.

Reply:
I think allowing multiple mentions is fine, but I find this user experience more convenient because it allows a user to easily identify who has not been mentioned in a message. If you want to mention a user twice, it would make more sense just to use their name directly, for example:

```
@David @Jupyternaut Can you take a look at the changes to `src/chat-commands/`?

David, please also check if the APIs being used are compatible with XYZ
```